commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/RecordDefinitionTranslator.java

```diff
@@ -0,0 +1,733 @@
+    final private static int    type_qstring       = 30;         // Convert to string
+
+    private static FieldType getFieldType(int typeID)
+    {
+        int type = typeID & TYPE_ID_MASK;
+        switch (type)
+        {
+            case type_boolean:
+                return FieldType.BOOLEAN;
+            case type_swapint:
+            case type_biasedswapint:
+            case type_keyedint:
+            case type_int:
+            case type_filepos:
+                return FieldType.INTEGER;
+            case type_real:
+                return FieldType.REAL;
+            case type_decimal:
+                return FieldType.DECIMAL;
+            case type_data:
+                return FieldType.BINARY;
+            case type_char:
+                return FieldType.CHAR;
+            case type_utf8:
+            case type_string:
+            case type_qstring:
+            case type_unicode:
+                return FieldType.STRING;
+            case type_varstring:
+            case type_varunicode:
+                return FieldType.VAR_STRING;
+            case type_set:
+                return FieldType.SET;
+            case type_record:
+                return FieldType.RECORD;
+            case type_table:
+                return FieldType.DATASET;
+            default:
+                return FieldType.UNKNOWN;
+        }
+    }
+
+    private static HpccSrcType getSourceType(int typeID)
+    {
+        int type = typeID & TYPE_ID_MASK;
+        switch (type)
+        {
+            case type_int:
+            case type_real:
+                return HpccSrcType.LITTLE_ENDIAN;
+            case type_swapint:
+            case type_biasedswapint:
+            case type_keyedint:
+            case type_filepos:
+                return HpccSrcType.BIG_ENDIAN;
+            case type_utf8:
+                return HpccSrcType.UTF8;
+            case type_char:
+            case type_string:
+            case type_varstring:
+                return HpccSrcType.SINGLE_BYTE_CHAR;
+            case type_qstring:
+                return HpccSrcType.QSTRING;
+            case type_unicode:
+            case type_varunicode:
+                return HpccSrcType.UTF16LE;
+            default:
+                return HpccSrcType.UNKNOWN;
+        }
+    }
+
+    private static boolean isFixedLength(int typeID)
+    {
+        int varLengthBit = typeID & FLAG_UNKNOWN_SIZE;
+        return varLengthBit == 0;
+    }
+
+    private static boolean isUnsigned(int typeID)
+    {
+        return (typeID & FLAG_UNSIGNED) != 0;
+    }
+
+    /**
+     * toECL Converts the provided FieldDef into an ECL record definition
+     *
+     * @param field
+     *            the FieldDef to convert
+     * @return ECL Record defintion as a String
+     * @throws Exception
+     */
+    public static String toECL(FieldDef field) throws Exception
+    {
+        if (field.getFieldType() != FieldType.RECORD)
```

Comment:
does this imply that all FieldDefs are required to be of type Record?
Does that mean that FieldDef should actually be ECLRecord? or maybe this method should be "toECLRecord" ?